Sources/App/WebView/LocalAuthentication/AuthenticationService.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+import LocalAuthentication
+import UIKit
+
+protocol AuthenticationServiceProtocol {
+    func authenticate()
+}
+
+class AuthenticationService: AuthenticationServiceProtocol {
+    private let context = LAContext()
+
+    func authenticate() {
+        var error: NSError?
+        if context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error) {
+            authenticate(policy: .deviceOwnerAuthenticationWithBiometrics)
+        } else if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+            authenticate(policy: .deviceOwnerAuthentication)
+        } else {
+            print("Failed to authenticate")
+        }
+    }
+
+    private func authenticate(policy: LAPolicy) {
+        context.evaluatePolicy(policy, localizedReason: "Authentication required") { approved, error in
```

Comment:
the hard part here is also blocking the ui while this should be happening, and e.g. in the app switcher. how would you imagine this to be implemented?

Reply:
Usually most apps blur the UI using and overlay UIView, what do you think? I'd go for that, although at least in the simulator I wasn't able to dismiss iOS biometric/password screen (which is visible in the video), so maybe it is already doing that.

Reply:
Something like this, I just added to the PR, but Im not 100% sure the App Switcher will show this overlay if the user has already authenticated.
![Simulator Screenshot - iPhone SE (3rd generation) - 2023-08-21 at 22 44 06](https://github.com/home-assistant/iOS/assets/5808343/da260a68-89cd-4f20-88e5-2fb916287777)